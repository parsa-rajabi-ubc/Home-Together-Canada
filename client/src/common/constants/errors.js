/**
 * @Author:     Rachelle Gelden
 * @Created:    2021.02.12
 *
 * @Description: Constants for common error messages that are used in multiple places
 *
 */

export const SESSION_ERR = 'There was error with your session.';
