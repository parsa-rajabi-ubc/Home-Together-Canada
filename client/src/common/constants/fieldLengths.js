export const DEFAULT_STRING_LENGTH = 255;

export const USER_FIELD_LENGTHS = {
    USERNAME: 50,
    PASSWORD: DEFAULT_STRING_LENGTH,
    EMAIL: DEFAULT_STRING_LENGTH,
    FIRST_NAME: 100,
    LAST_NAME: 100,
    PHONE_NUMBER: 12,
    ADDRESS_LINE_1: 100,
    ADDRESS_LINE_2: 100,
    CITY: 60,
    PROVINCE: 2,
    POSTAL_CODE: 6,
    MAILING_ADDRESS_LINE_1: 100,
    MAILING_ADDRESS_LINE_2: 100,
    MAILING_CITY: 60,
    MAILING_PROVINCE: 2,
    MAILING_POSTAL_CODE: 6
};

export const MEMBER_FIELD_LENGTHS = {
    GENDER: 50,
    GENDER_DESCRIPTION: 100,
    STATUS: DEFAULT_STRING_LENGTH,
    HAS_HOME_TO_SHARE_DESCRIPTION: 100,
    INTERESTED_IN_BUYING_HOME_DESCRIPTION: 100,
    RELIGION_DESCRIPTION: 100,
    DIET_DESCRIPTION: 100,
    HEALTH_MOBILITY_DESCRIPTION: 100,
    ALLERGIES_DESCRIPTION: 100,
    PETS_DESCRIPTION: 100,
    SMOKING_DESCRIPTION: 100,
    BIO: 2000,
    WORK_STATUS: DEFAULT_STRING_LENGTH,
    DEACTIVATION_REASON: 500
};

export const BUSINESS_FIELD_LENGTHS = {
    BUSINESS_NAME: 100,
    INCORPORATED_OWNERS_NAMES: DEFAULT_STRING_LENGTH,
    MAP_ADDRESS_LINE_1: 100,
    MAP_ADDRESS_LINE_2: 100,
    MAP_CITY: 60,
    MAP_PROVINCE: 2,
    MAP_POSTAL_CODE: 6,
    WEBSITE: DEFAULT_STRING_LENGTH
}

export const LISTING_FIELD_LENGTHS = {
    TITLE: 75,
    SHORT_DESCRIPTION: DEFAULT_STRING_LENGTH,
    FULL_DESCRIPTION: 2000,
    CONTACT_NAME: 200,
    RATES_AND_FEES: DEFAULT_STRING_LENGTH,
    ADDRESS_LINE_1: 100,
    ADDRESS_LINE_2: 100,
    CITY: 60,
    PROVINCE: 2,
    POSTAL_CODE: 6,
    EVENT_DATE_TIME: DEFAULT_STRING_LENGTH,
    ORDER_ID: DEFAULT_STRING_LENGTH
}

export const MESSAGE_FIELD_LENGTHS = {
    SENDER_USERNAME: 50,
    RECEIVER_USERNAME: 50,
    CONTENT: 2000
}
