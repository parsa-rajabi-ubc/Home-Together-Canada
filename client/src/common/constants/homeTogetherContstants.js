/**
 * @Author:     Rachelle Gelden
 * @Created:    2020.12.20
 *
 * @Description: constants related to the Home Together Canada organization
 *
 */

export const HOME_TOGETHER_EMAIL = 'info@hometogether.ca';