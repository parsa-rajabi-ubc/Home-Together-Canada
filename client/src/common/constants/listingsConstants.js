/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.02.04
 *
 * @Description: Constants for Listings
 *
 */


export const CREATE_LISTINGS_TEXT = "Create Listing"
export const SERVICES_TEXT = "Home Sharing Services";
export const CLASSIFIEDS_TEXT = "Local Classifieds";

export const SHORT_DESC_CHAR_COUNT = 50;