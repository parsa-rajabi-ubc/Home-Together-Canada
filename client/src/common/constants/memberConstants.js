/**
 * @Author:     Rachelle Gelden
 * @Created:    2020.02.01
 *
 * @Description: Constants for members
 *
 */

export const STATUSES = {
    SINGLE: 'Single',
    COUPLE: 'Couple',
    COUPLE_WITH_CHILDREN: 'Couple With Children',
    SINGLE_PARENT: 'Single Parent',
    EXISTING_GROUP: 'Existing Group'
}

export const MAX_AREAS_OF_INTEREST = 5;
