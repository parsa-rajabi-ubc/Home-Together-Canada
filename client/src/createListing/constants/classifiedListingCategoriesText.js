/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.02.04
 *
 * @Description: Constants for Classifieds Text
 *
 */

const OTHER = "Other";

export const BUSINESS_CLASSIFIEDS_CATEGORIES = {
    RENTALS: "Rentals",
    HOUSE_YARD: "House & Yard Services",
    LEGAL_SALES: "Legal & Sales",
    CLASSES_CLUBS: "Classes, Clubs & Events",
}

export const RENTALS_SUBCATEGORIES = {
    HOUSE: "House",
    PARTIAL_HOUSE: "Partial house / suite up or down",
    DUPLEX: "Duplex / triplex",
    APART: "Apartment",
    CONDO: "Condo/townhouse",
    RURAL: "Rural",
    COUNTRY: "Country",
    OTHER: OTHER
}

export const HOUSE_YARD_SERVICES_SUBCATEGORIES = {
    CLEAN_CARPET: "Carpet Cleaning",
    COMPUTER_TECH: "Computer & Technical Services",
    DELIVERY: "Delivery Services",
    DUCT_CLEANING: "Duct Cleaning",
    GARDENING: "Gardening, Yard Work & Yard Maintenance",
    HANDYMAN: "Handyman Services",
    HOUSE_KEEPER: "Housekeeper/Cleaner",
    HOME_CARE: "Home Care Services",
    HOUSE_MAINTENANCE: "House Maintenance & Repair",
    MEAL_SERVICE: "Meal Services",
    MOVING: "Moving & Storage",
    ORGANIZING: "Organizing/De-cluttering",
    PET: "Pet Services",
    RIDE: "Ride Sharing & Transportation",
    CLEAN_WINDOW: "Window Cleaning",
    OTHER: OTHER
}

export const LEGAL_SALES_AGENCIES_SUBCATEGORIES = {
    LEGALITIES: "Multi-Ownership Purchasing: Legalities",
    REALTORS: "Multi-Ownership Purchasing: Realtors",
    MULTI_SHARE_HOME: "Multi-Ownership/Tenant/Shared Home Insurance",
    SHARE_CARE: "Shared Vehicle Insurance & Legalities",
    TAX: "Understanding Home Sharing & Taxes",
    MULTI_TENANT: "Multi-Tenant Rentals and Leases",
    SHARED_HOME: "Shared Homes A-Z",
    OTHER: OTHER
}

export const CLASSES_CLUBS_EVENTS_SUBCATEGORIES = {
    CLASSES: "Classes & Lessons (Art, Music, Computer, Cooking, Language, etc)",
    EVENTS: "Events & Activities (Paint-night, Art Show, Wine Tasting, Concert, etc)",
    EXERCISE: "Exercise Classes (Yoga, Pilates, Walking, Running, etc.)",
    CLUBS: "Groups & Clubs (Computer, Bird Watching, Book Club, Singing, etc)",
    OTHER: OTHER
}