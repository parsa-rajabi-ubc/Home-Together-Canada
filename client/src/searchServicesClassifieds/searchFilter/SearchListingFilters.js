/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.18
 *
 * @Description: Search Filter Component
 *
 */

import React, {useContext} from 'react';
import PropTypes from "prop-types";
import SubmitButton from "../../common/forms/SubmitButton";
import Dropdown from "../../common/forms/Dropdown";
import {dropdownDefaultCSS, dropdownErrorCSS} from "../../css/dropdownCSSUtil";
import {listingContext} from "../SearchListingContainer";
import Checkbox from "../../common/forms/Checkbox";
import {CREATE_LISTING_CONTROLS_TEXT} from "../../createListing/CreateListingControls";
import SearchArea from "../../common/listings/SearchArea";
import {MEMBER_SERVICE_CATEGORIES} from "../../createListing/constants/serviceListingCategoriesText";

const TEXT = {
    DROPDOWN: CREATE_LISTING_CONTROLS_TEXT.SELECT_LISTING_CATEGORY,
    CHECKBOX: "Select Subcategories",
}

function SearchListingFilters(props) {
    const {
        handleSearchAreaChange,
        searchArea,

        categoryOptions,
        selectedCategory,
        subcategories,

        handleSubcategoriesChange,
        handleCategoryChange,

        selectedCategoryError,
        selectedSubcategoryError,

        searchAreaProvinceError,
        searchAreaCityError,
        searchAreaRadiusError,

        onSubmit,
    } = props;

    const listingPage = useContext(listingContext);

    const subcategoryCheckboxes = subcategories.map(
        (subcategory, index) =>
            <Checkbox
                key={subcategory + index}
                label={subcategory.toString()}
                id={subcategory.toString()}
                fontNormal={true}
                onChange={handleSubcategoriesChange}
            />
    );

    return (
        <div className="col-start-1 col-end-7 py-5 px-5 m-6 bg-white shadow-lg rounded-xl">
            <div className="grid grid-cols-2 gap-6">
                <div className="col-span-3 sm:col-span-2">
                    <label className="text-2xl font-medium">Search Filtering</label>
                    <SubmitButton
                        inputValue={"Search"}
                        className="btn btn-green w-1/4 float-right py-2"
                        onClick={onSubmit}
                    />

                    <section className={"my-8"} key={listingPage}>
                        <section className={"mb-4"}>
                            <label className="label text-lg mb-1">Search Area</label>
                            <SearchArea
                                searchArea={searchArea}
                                onChange={handleSearchAreaChange}
                                searchAreaProvinceError={searchAreaProvinceError}
                                searchAreaCityError={searchAreaCityError}
                                searchAreaRadiusError={searchAreaRadiusError}
                            />
                        </section>

                        <label className="label text-lg mb-1">{TEXT.DROPDOWN}</label>
                        <Dropdown
                            options={categoryOptions}
                            onChange={handleCategoryChange}
                            dropdownCSS={selectedCategoryError ? dropdownErrorCSS : dropdownDefaultCSS}
                        />
                    </section>

                    {(selectedCategory && selectedCategory !== MEMBER_SERVICE_CATEGORIES.MEMBER_HOME)  &&
                    <section className={`${selectedSubcategoryError && "pl-1 border rounded-lg border-red-500"} my-4`}>
                        <p className={"label text-lg mb-1"}>{TEXT.CHECKBOX}</p>
                        {subcategoryCheckboxes}
                    </section>
                    }

                </div>

            </div>
        </div>
    );
}

SearchListingFilters.propTypes = {
    handleSearchAreaChange: PropTypes.func.isRequired,
    searchArea: PropTypes.shape({
        province: PropTypes.string,
        city: PropTypes.string,
        radius: PropTypes.number
    }),

    categoryOptions: PropTypes.array.isRequired,
    selectedCategory: PropTypes.string,
    subcategories: PropTypes.array.isRequired,

    handleSubcategoriesChange: PropTypes.func.isRequired,
    handleCategoryChange: PropTypes.func.isRequired,

    selectedCategoryError: PropTypes.bool,
    selectedSubcategoryError: PropTypes.bool,

    searchAreaProvinceError: PropTypes.bool,
    searchAreaCityError: PropTypes.bool,
    searchAreaRadiusError: PropTypes.bool,

    onSubmit: PropTypes.func.isRequired,
};

export default SearchListingFilters;