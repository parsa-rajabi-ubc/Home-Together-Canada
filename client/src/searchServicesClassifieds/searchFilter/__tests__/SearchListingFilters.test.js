/**
 * @Author:     Parsa Rajabi
 * @Created:    2021.2.18
 *
 * @Description: Search Listing Filter Container Test
 *
 */

import React from 'react';
// import renderer from  'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import SearchListingFilters from "../SearchListingFilters";
import {BUSINESS_CLASSIFIEDS_CATEGORIES} from "../../../createListing/constants/classifiedListingCategoriesText";
import {
    BUSINESS_SERVICE_CATEGORIES,
} from "../../../createListing/constants/serviceListingCategoriesText";
import {
    BUSINESS_CLASSIFIEDS_CATEGORIES_DROPDOWN,
    RENTALS_SUBCATEGORIES_LIST
} from "../../../createListing/constants/classifiedListingConstants";
import {
    BUSINESS_SERVICE_CATEGORIES_DROPDOWN,
    CO_HOUSING_SUBCATEGORIES_LIST
} from "../../../createListing/constants/serviceListingConstants";

describe('SearchListingFilters', () => {
    describe('Container test', () => {
        it('should match snapshot test with Classifieds Category with Errors set to true', () => {
            //given
            const props = {
                handleSearchAreaChange: jest.fn(),

                categoryOptions: BUSINESS_CLASSIFIEDS_CATEGORIES_DROPDOWN,
                selectedCategory: BUSINESS_CLASSIFIEDS_CATEGORIES.RENTALS,
                subcategories: [RENTALS_SUBCATEGORIES_LIST],

                handleSubcategoriesChange: jest.fn(),
                handleCategoryChange: jest.fn(),

                selectedCategoryError: true,
                selectedSubcategoryError: true,
                searchAreaProvinceError: true,
                searchAreaCityError: true,
                searchAreaRadiusError: true,

                onSubmit: jest.fn(),
            }

            //when
            const renderer = new ShallowRenderer();
            renderer.render(<SearchListingFilters {...props}/>);
            const tree = renderer.getRenderOutput();

            //then
            expect(tree).toMatchSnapshot();
        });
        it('should match snapshot test with Services Category with Errors set to true', () => {
            //given
            const props = {
                handleSearchAreaChange: jest.fn(),

                categoryOptions: BUSINESS_SERVICE_CATEGORIES_DROPDOWN,
                selectedCategory: BUSINESS_SERVICE_CATEGORIES.CO_HOUSING,
                subcategories: [CO_HOUSING_SUBCATEGORIES_LIST],

                handleSubcategoriesChange: jest.fn(),
                handleCategoryChange: jest.fn(),

                selectedCategoryError: true,
                selectedSubcategoryError: true,
                searchAreaProvinceError: true,
                searchAreaCityError: true,
                searchAreaRadiusError: true,

                onSubmit: jest.fn(),
            }

            //when
            const renderer = new ShallowRenderer();
            renderer.render(<SearchListingFilters {...props}/>);
            const tree = renderer.getRenderOutput();

            //then
            expect(tree).toMatchSnapshot();
        });
        it('should match snapshot test with Classifieds Category with Errors set to false', () => {
            //given
            const props = {
                handleSearchAreaChange: jest.fn(),

                categoryOptions: BUSINESS_CLASSIFIEDS_CATEGORIES_DROPDOWN,
                selectedCategory: BUSINESS_CLASSIFIEDS_CATEGORIES.RENTALS,
                subcategories: [RENTALS_SUBCATEGORIES_LIST],

                handleSubcategoriesChange: jest.fn(),
                handleCategoryChange: jest.fn(),

                selectedCategoryError: false,
                selectedSubcategoryError: false,
                searchAreaProvinceError: false,
                searchAreaCityError: false,
                searchAreaRadiusError: false,

                onSubmit: jest.fn(),
            }

            //when
            const renderer = new ShallowRenderer();
            renderer.render(<SearchListingFilters {...props}/>);
            const tree = renderer.getRenderOutput();

            //then
            expect(tree).toMatchSnapshot();
        });
        it('should match snapshot test with Services Category with Errors set to false', () => {
            //given
            const props = {
                handleSearchAreaChange: jest.fn(),

                categoryOptions: BUSINESS_SERVICE_CATEGORIES_DROPDOWN,
                selectedCategory: BUSINESS_SERVICE_CATEGORIES.CO_HOUSING,
                subcategories: [CO_HOUSING_SUBCATEGORIES_LIST],

                handleSubcategoriesChange: jest.fn(),
                handleCategoryChange: jest.fn(),

                selectedCategoryError: false,
                selectedSubcategoryError: false,
                searchAreaProvinceError: false,
                searchAreaCityError: false,
                searchAreaRadiusError: false,

                onSubmit: jest.fn(),
            }

            //when
            const renderer = new ShallowRenderer();
            renderer.render(<SearchListingFilters {...props}/>);
            const tree = renderer.getRenderOutput();

            //then
            expect(tree).toMatchSnapshot();
        });
    });
});