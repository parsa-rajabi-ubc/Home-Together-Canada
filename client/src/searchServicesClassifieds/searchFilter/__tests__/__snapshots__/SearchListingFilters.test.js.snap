// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchListingFilters Container test should match snapshot test with Classifieds Category with Errors set to false 1`] = `
<div
  className="col-start-1 col-end-7 py-5 px-5 my-6 ml-6 bg-white shadow-lg rounded-xl"
>
  <div
    className="grid grid-cols-2 gap-6"
  >
    <div
      className="col-span-3 sm:col-span-2"
    >
      <label
        className="text-2xl font-medium"
      >
        Search Filtering
      </label>
      <SubmitButton
        className="btn btn-green w-1/4 float-right py-2"
        inputValue="Search"
        onClick={[MockFunction]}
      />
      <section
        className="my-8"
      >
        <section
          className="mb-4"
        >
          <label
            className="label text-lg mb-1"
          >
            Search Area
          </label>
          <SearchArea
            onChange={[MockFunction]}
            searchAreaCityError={false}
            searchAreaProvinceError={false}
            searchAreaRadiusError={false}
          />
        </section>
        <label
          className="label text-lg mb-1"
        >
          Select Listing Category
        </label>
        <Dropdown
          dropdownCSS={
            Object {
              "control": [Function],
              "menuPortal": [Function],
            }
          }
          initialSelection={
            Object {
              "label": "Rentals",
              "value": "Rentals",
            }
          }
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "label": "Rentals",
                "value": "Rentals",
              },
              Object {
                "label": "House & Yard Services",
                "value": "House & Yard Services",
              },
              Object {
                "label": "Legal & Sales",
                "value": "Legal & Sales",
              },
              Object {
                "label": "Classes, Clubs & Events",
                "value": "Classes, Clubs & Events",
              },
            ]
          }
        />
      </section>
      <section
        className="false my-4"
      >
        <p
          className="label text-lg mb-1"
        >
          Select Subcategories
        </p>
        <Checkbox
          fontNormal={true}
          id="House,Partial house / suite up or down,Duplex / triplex,Apartment,Condo/townhouse,Rural,Country,Other"
          label="House,Partial house / suite up or down,Duplex / triplex,Apartment,Condo/townhouse,Rural,Country,Other"
          onChange={[MockFunction]}
        />
      </section>
    </div>
  </div>
</div>
`;

exports[`SearchListingFilters Container test should match snapshot test with Classifieds Category with Errors set to true 1`] = `
<div
  className="col-start-1 col-end-7 py-5 px-5 my-6 ml-6 bg-white shadow-lg rounded-xl"
>
  <div
    className="grid grid-cols-2 gap-6"
  >
    <div
      className="col-span-3 sm:col-span-2"
    >
      <label
        className="text-2xl font-medium"
      >
        Search Filtering
      </label>
      <SubmitButton
        className="btn btn-green w-1/4 float-right py-2"
        inputValue="Search"
        onClick={[MockFunction]}
      />
      <section
        className="my-8"
      >
        <section
          className="mb-4"
        >
          <label
            className="label text-lg mb-1"
          >
            Search Area
          </label>
          <SearchArea
            onChange={[MockFunction]}
            searchAreaCityError={true}
            searchAreaProvinceError={true}
            searchAreaRadiusError={true}
          />
        </section>
        <label
          className="label text-lg mb-1"
        >
          Select Listing Category
        </label>
        <Dropdown
          dropdownCSS={
            Object {
              "control": [Function],
              "menuPortal": [Function],
            }
          }
          initialSelection={
            Object {
              "label": "Rentals",
              "value": "Rentals",
            }
          }
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "label": "Rentals",
                "value": "Rentals",
              },
              Object {
                "label": "House & Yard Services",
                "value": "House & Yard Services",
              },
              Object {
                "label": "Legal & Sales",
                "value": "Legal & Sales",
              },
              Object {
                "label": "Classes, Clubs & Events",
                "value": "Classes, Clubs & Events",
              },
            ]
          }
        />
      </section>
      <section
        className="pl-1 border rounded-lg border-red-500 my-4"
      >
        <p
          className="label text-lg mb-1"
        >
          Select Subcategories
        </p>
        <Checkbox
          fontNormal={true}
          id="House,Partial house / suite up or down,Duplex / triplex,Apartment,Condo/townhouse,Rural,Country,Other"
          label="House,Partial house / suite up or down,Duplex / triplex,Apartment,Condo/townhouse,Rural,Country,Other"
          onChange={[MockFunction]}
        />
      </section>
    </div>
  </div>
</div>
`;

exports[`SearchListingFilters Container test should match snapshot test with Services Category with Errors set to false 1`] = `
<div
  className="col-start-1 col-end-7 py-5 px-5 my-6 ml-6 bg-white shadow-lg rounded-xl"
>
  <div
    className="grid grid-cols-2 gap-6"
  >
    <div
      className="col-span-3 sm:col-span-2"
    >
      <label
        className="text-2xl font-medium"
      >
        Search Filtering
      </label>
      <SubmitButton
        className="btn btn-green w-1/4 float-right py-2"
        inputValue="Search"
        onClick={[MockFunction]}
      />
      <section
        className="my-8"
      >
        <section
          className="mb-4"
        >
          <label
            className="label text-lg mb-1"
          >
            Search Area
          </label>
          <SearchArea
            onChange={[MockFunction]}
            searchAreaCityError={false}
            searchAreaProvinceError={false}
            searchAreaRadiusError={false}
          />
        </section>
        <label
          className="label text-lg mb-1"
        >
          Select Listing Category
        </label>
        <Dropdown
          dropdownCSS={
            Object {
              "control": [Function],
              "menuPortal": [Function],
            }
          }
          initialSelection={
            Object {
              "label": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
              "value": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
            }
          }
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "label": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
                "value": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
              },
              Object {
                "label": "Home Share Facilitation & Supporting Services",
                "value": "Home Share Facilitation & Supporting Services",
              },
              Object {
                "label": "Home Sharing Businesses, Groups and Organizations",
                "value": "Home Sharing Businesses, Groups and Organizations",
              },
              Object {
                "label": "Government & Human Service Agencies Offering Shared Living & Support Services",
                "value": "Government & Human Service Agencies Offering Shared Living & Support Services",
              },
            ]
          }
        />
      </section>
      <section
        className="false my-4"
      >
        <p
          className="label text-lg mb-1"
        >
          Select Subcategories
        </p>
        <Checkbox
          fontNormal={true}
          id="Co-housing Groups & Communities,Cooperatives,Communal Living,Eco villages,Intergenerational Living,Shared Living First Step Homes,Other"
          label="Co-housing Groups & Communities,Cooperatives,Communal Living,Eco villages,Intergenerational Living,Shared Living First Step Homes,Other"
          onChange={[MockFunction]}
        />
      </section>
    </div>
  </div>
</div>
`;

exports[`SearchListingFilters Container test should match snapshot test with Services Category with Errors set to true 1`] = `
<div
  className="col-start-1 col-end-7 py-5 px-5 my-6 ml-6 bg-white shadow-lg rounded-xl"
>
  <div
    className="grid grid-cols-2 gap-6"
  >
    <div
      className="col-span-3 sm:col-span-2"
    >
      <label
        className="text-2xl font-medium"
      >
        Search Filtering
      </label>
      <SubmitButton
        className="btn btn-green w-1/4 float-right py-2"
        inputValue="Search"
        onClick={[MockFunction]}
      />
      <section
        className="my-8"
      >
        <section
          className="mb-4"
        >
          <label
            className="label text-lg mb-1"
          >
            Search Area
          </label>
          <SearchArea
            onChange={[MockFunction]}
            searchAreaCityError={true}
            searchAreaProvinceError={true}
            searchAreaRadiusError={true}
          />
        </section>
        <label
          className="label text-lg mb-1"
        >
          Select Listing Category
        </label>
        <Dropdown
          dropdownCSS={
            Object {
              "control": [Function],
              "menuPortal": [Function],
            }
          }
          initialSelection={
            Object {
              "label": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
              "value": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
            }
          }
          onChange={[MockFunction]}
          options={
            Array [
              Object {
                "label": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
                "value": "Co-housing, Co-ops, Intergenerational & Planned Neighbourhoods",
              },
              Object {
                "label": "Home Share Facilitation & Supporting Services",
                "value": "Home Share Facilitation & Supporting Services",
              },
              Object {
                "label": "Home Sharing Businesses, Groups and Organizations",
                "value": "Home Sharing Businesses, Groups and Organizations",
              },
              Object {
                "label": "Government & Human Service Agencies Offering Shared Living & Support Services",
                "value": "Government & Human Service Agencies Offering Shared Living & Support Services",
              },
            ]
          }
        />
      </section>
      <section
        className="pl-1 border rounded-lg border-red-500 my-4"
      >
        <p
          className="label text-lg mb-1"
        >
          Select Subcategories
        </p>
        <Checkbox
          fontNormal={true}
          id="Co-housing Groups & Communities,Cooperatives,Communal Living,Eco villages,Intergenerational Living,Shared Living First Step Homes,Other"
          label="Co-housing Groups & Communities,Cooperatives,Communal Living,Eco villages,Intergenerational Living,Shared Living First Step Homes,Other"
          onChange={[MockFunction]}
        />
      </section>
    </div>
  </div>
</div>
`;
